src/plugins/kibana_utils/docs/state_sync/README.md

```diff
@@ -3,6 +3,18 @@
 State syncing utilities are a set of helpers for syncing your application state
 with URL or browser storage.
 
+**When you should consider using state syncing utils:**
+
+- You want to sync your application state with URL in similar manner analyze applications do that.
+- You want to follow platform's <<history-and-location, working with browser history and location best practices>> out of the box.
+- You want to support `state:storeInSessionStore` escape hatch for URL overflowing out of the box.
```

Comment:
So `state:storeInSessionStore` relies on _a and _g parameters in the URL?

Reply:
No, not really. 

When using state syncing utils (with `kbnUrlStateStorage` to be precise) you can configure your state to be synced to any query key. You can also set it up to sync only hash into URL. This is covered in docs of `stateSync`, which is linked from this general guide: https://github.com/elastic/kibana/blob/master/src/plugins/kibana_utils/docs/state_sync/storages/kbn_url_storage.md#setting-url-format-option